test/python/test_updateconfig.py

```diff
@@ -46,3 +46,23 @@ def test_update_config(self):
         ngraph_optimizer.parameter_map["ice_cores"].s = b'12'
         assert config_new.__str__(
         ) == 'allow_soft_placement: true\ngraph_options {\n  rewrite_options {\n    meta_optimizer_iterations: ONE\n    min_graph_nodes: -1\n    custom_optimizers {\n      name: "ngraph-optimizer"\n      parameter_map {\n        key: "device_id"\n        value {\n          s: ""\n        }\n      }\n      parameter_map {\n        key: "ice_cores"\n        value {\n          s: "12"\n        }\n      }\n      parameter_map {\n        key: "max_batch_size"\n        value {\n          s: "64"\n        }\n      }\n      parameter_map {\n        key: "ngraph_backend"\n        value {\n          s: "CPU"\n        }\n      }\n    }\n  }\n}\n'
+
+    @pytest.mark.skipif(
+        not ngraph_bridge.is_grappler_enabled(), reason='Only for Grappler')
+    def test_update_config_adds_optimizer_only_once(self):
+
+        # Helper function to count the number of occurances in a config
+        def count_ng_optimizers(config):
+            custom_opts = config.graph_options.rewrite_options.custom_optimizers
+            count = 0
+            for i in range(len(custom_opts)):
+                if custom_opts[i].name == 'ngraph-optimizer':
+                    count += 1
+            return count
+
+        config = tf.ConfigProto(allow_soft_placement=True)
```

Comment:
We should remove the `allow_soft_placement` as it's not needed.

Reply:
This reflects a real world config, where the user's config might have extra information in it. We have pytests for update_config that check that "allow_soft_placement" remains in place after passing through update_config.

In this particular case we are not rechecking if `allow_soft_placement` is present in the output. However `allow_soft_placement` is there as a representation of a real world non empty config

Reply:
Ok - that makes sense. Then please add a comment in the code to reflect that. Remember - git hub review comments go away - but the comments in the code aren't.

Reply:
Added comments